client/src/controllers/MessagesController.ts

```diff
@@ -0,0 +1,77 @@
+import { Controller } from '@hotwired/stimulus';
+
+/**
+ * Adds the ability for a controlled element to pick messages from template,
+ * add/stack (depending upon clear:true/false) them within the provided unordered list container
+ * when called.
+ *
+ * @example
+ * <div data-controller="w-messages"
+ *      data-action="w-messages:add@document->w-messages#add"
+        data-w-messages-added-class="new"
+        data-w-messages-show-class="appear"
+ * >
+      <ul data-w-messages-target="messages"></ul>
+      <template data-w-messages-target="templates">
+        <li data-message-status="error/success"><span></span></li>
+      </template>
+ * </div>
+ */
+export class MessagesController extends Controller<HTMLFormElement> {
+  static classes = ['added', 'show'];
+  static targets = ['messages', 'templates'];
+  static values = {
+    showDelay: { default: 100, type: Number },
+  };
+
+  declare readonly hasMessagesTarget: boolean;
+  declare readonly hasAddedClass: boolean;
+  declare readonly hasShowClass: boolean;
+  declare readonly hasTemplatesTarget: boolean;
+  declare messagesTarget: HTMLElement;
+  declare readonly addedClass: string;
+  declare readonly showClass: string;
+  declare readonly showDelayValue: number;
+  declare templatesTarget: HTMLTemplateElement;
+
+  add(
+    event?: CustomEvent<{
+      /** Message status level, based on Django's message types. */
+      status?: 'success' | 'error' | 'warning' | string;
+      /** Content for the message, HTML not supported. */
+      text?: string;
+      /** flag for clearing or stacking messages */
+      clear?: boolean;
+    }>,
+  ) {
+    const {
+      status = 'other',
+      text: statusText,
+      clear = false,
+    } = event?.detail || {};
+    if (!status && !statusText) return;
+
+    if (this.hasAddedClass) {
+      this.element.classList.add(this.addedClass);
+    }
+
+    const message = this.templatesTarget.content
+      .querySelector(`[data-message-status="${status}"]`)
+      ?.cloneNode(true);
+    if (message) {
+      const spanElement = (message as HTMLElement).querySelector(
+        'span',
+      ) as HTMLElement;
+      spanElement.textContent = statusText as string;
+      if (this.hasMessagesTarget) {
+        if (clear) this.messagesTarget.innerHTML = '';
+        this.messagesTarget.appendChild(message);
+      }
+    }
+    if (this.hasShowClass) {
+      setTimeout(() => {
+        this.element.classList.add(this.showClass);
+      }, this.showDelayValue);
+    }
+  }
```

Comment:
```suggestion

    if (clear) this.clear();

    const templateContent = this.templatesTarget.content;

    const message = (
      templateContent.querySelector(`[data-message-status="${status}"]`) ||
      templateContent.firstElementChild
    )?.cloneNode(true);

    if (message instanceof HTMLElement) {
      const textElement = message.lastElementChild;

      if (textElement instanceof HTMLElement) {
        textElement.textContent = text;
      }

      this.containerTarget.appendChild(message);

      if (this.hasShowClass) {
        setTimeout(() => {
          this.element.classList.add(this.showClass);
        }, this.showDelayValue);
      }
    }
  }

  clear() {
    this.containerTarget.innerHTML = '';
  }
```

Let's do a  bit  of clean up here also

* pull out `clean` into it's own method, this is very useful to maybe be able to call in isolation
* Only add the shown class if we actually add a message
* Using `instanceof` we can avoid a few of the whole `as something` declarations, it's also a more correct way to do this in TypeScript and avoids errors
* As noted below, if we cannot find a matching status template, just grab the FIRST element child instead to use
* Avoid too much naming and wording that implies we are using a specific DOM element, that's up to the HTML
* Finally, instead of grabbing the `span`, lets' grab the lastElementChild and add content there, this allows for some nice approaches like using `strong` instead.